e2e-tests/src/driver.rs

```diff
@@ -95,133 +103,121 @@ impl Driver {
+                    ProtocolSandboxConfig::Icp(_) => Protocol::Icp,
+                    ProtocolSandboxConfig::Ethereum(_) => Protocol::Ethereum,
+                };
+
+                if &config_protocol != protocol {
+                    continue;
+                }
+
+                let sandbox_env = match sandbox_cfg {
+                    ProtocolSandboxConfig::Stellar(config) => ProtocolSandboxEnvironment::Stellar(
+                        StellarSandboxEnvironment::init(config.clone())?,
+                    ),
+                    ProtocolSandboxConfig::Near(config) => ProtocolSandboxEnvironment::Near(
+                        NearSandboxEnvironment::init(config.clone()).await?,
+                    ),
+                    ProtocolSandboxConfig::Icp(config) => ProtocolSandboxEnvironment::Icp(
+                        IcpSandboxEnvironment::init(config.clone())?,
+                    ),
+                    ProtocolSandboxConfig::Ethereum(config) => {
+                        ProtocolSandboxEnvironment::Ethereum(EthereumSandboxEnvironment::init(
+                            config.clone(),
+                        )?)
+                    }
+                };
+
+                initialized_protocols.insert(*protocol, sandbox_env);
+                break;
+            }
+        }
+
+        for protocol_name in &self.environment.protocols {
+            let protocol_path = protocols_dir.join(&protocol_name.to_string().to_lowercase());
+
+            if !protocol_path.is_dir() {
+                self.environment
+                    .output_writer
+                    .write_str(&format!("No directory for protocol: {}", protocol_name));
+                continue;
+            }
+
+            let Some(sandbox) = initialized_protocols.get(&protocol_name) else {
+                bail!("Sandbox not initialized for protocol: {}", protocol_name);
+            };
+
+            let mero = self.setup_mero(&vec![sandbox.clone()]).await?;
+
+            let Some((inviter, invitees)) = self.pick_inviter_node(&mero.ds) else {
+                bail!("Not enough nodes to run test for protocol {protocol_name}")
+            };
+
+            self.environment
+                .output_writer
+                .write_str(&format!("Picked inviter: {inviter}"));
+            self.environment
+                .output_writer
+                .write_str(&format!("Picked invitees: {invitees:?}"));
 
-        // Run scenarios directory by directory
-        let scenarios_dir = self.environment.input_dir.join("scenarios");
-        let mut entries = read_dir(scenarios_dir).await?;
+            let mut applications = read_dir(&protocol_path).await?;
+            while let Some(app) = applications.next_entry().await? {
+                let protocol_name = protocol_name.to_string().to_lowercase();
+                if app.file_type().await?.is_dir() {
+                    continue;
+                }
```

Comment:
why not [`is_file`](https://doc.rust-lang.org/nightly/std/fs/struct.FileType.html#method.is_file)?